apiserver/facades/agent/uniter/uniter.go

```diff
@@ -984,6 +984,111 @@ func (u *UniterAPI) WatchActionNotifications(args params.Entities) (params.Strin
 	return common.WatchActionNotifications(args, canAccess, watchOne), nil
 }
 
+// WatchActionNotifications returns a StringsWatcher for observing the status of an Upgrade Series
+func (u *UniterAPI) WatchUpgradeSeriesNotifications(args params.Entities) (params.NotifyWatchResults, error) {
+	result := params.NotifyWatchResults{
+		Results: make([]params.NotifyWatchResult, len(args.Entities)),
+	}
+	canAccess, err := u.accessUnit()
+	if err != nil {
+		return params.NotifyWatchResults{}, err
+	}
+	for i, entity := range args.Entities {
+		tag, err := names.ParseUnitTag(entity.Tag)
+		if err != nil {
+			result.Results[i].Error = common.ServerError(common.ErrPerm)
+			continue
+		}
+		watcherId := ""
+		if !canAccess(tag) {
+			result.Results[i].Error = common.ServerError(common.ErrPerm)
+			continue
+		}
+		unit, err := u.getUnit(tag)
+		if err != nil {
+			result.Results[i].Error = common.ServerError(err)
+			continue
+		}
+		w, err := unit.WatchUpgradeSeriesNotifications()
+		if err != nil {
+			result.Results[i].Error = common.ServerError(err)
+			continue
+		}
+		watcherId = u.resources.Register(w)
+		result.Results[i].NotifyWatcherId = watcherId
```

Comment:
Per your earlier question of "why do I have 2 events":
https://github.com/juju/juju/blob/develop/apiserver/common/watch.go#L39
"
	// Consume the initial event. Technically, API
	// calls to Watch 'transmit' the initial event
	// in the Watch response. But NotifyWatchers
	// have no state to transmit.
	if _, ok := <-watch.Changes(); ok {
"

Reply:
(put another way, the call to Watch() should return the watcher identity, and the content of the first Next() result, and then on the client side, those should be split up into the appropriate responses. A NotifyWatcher doesn't have any actual content to return just an event that gets consumed on the server and regenerated on the client)

Reply:
Hey @howbazaar, I swallowed the extra event on the server side,  here I was swallowing both the servers initial event and the clients initial event on the client. My fault.